lexer_schemas/validation.py

```diff
@@ -0,0 +1,61 @@
+import jsonschema
+from datetime import datetime
+from typing import Optional
+from pydantic import BaseModel, ValidationError
+from lexer_schemas import validation_helper
+
+json_schema_validator = {
+    "$ref": validation_helper.ref,
+    "additionalItems": validation_helper.additionalItems,
+    "additionalProperties": validation_helper.additionalProperties,
+    "allOf": validation_helper.allOf,
+    "anyOf": validation_helper.anyOf,
+    "const": validation_helper.const,
+    "contains": validation_helper.contains,
+    "dependencies": validation_helper.dependencies,
+    "enum": validation_helper.enum,
+    "exclusiveMaximum": validation_helper.exclusiveMaximum,
+    "exclusiveMinimum": validation_helper.exclusiveMinimum,
+    "format": validation_helper.format,
+    "if": validation_helper.if_,
+    "items": validation_helper.items,
+    "maxItems": validation_helper.maxItems,
+    "maxLength": validation_helper.maxLength,
+    "maxProperties": validation_helper.maxProperties,
+    "maximum": validation_helper.maximum,
+    "minItems": validation_helper.minItems,
+    "minLength": validation_helper.minLength,
+    "minProperties": validation_helper.minProperties,
+    "minimum": validation_helper.minimum,
+    "multipleOf": validation_helper.multipleOf,
+    "not": validation_helper.not_,
+    "oneOf": validation_helper.oneOf,
+    "pattern": validation_helper.pattern,
+    "patternProperties": validation_helper.patternProperties,
+    "properties": validation_helper.properties,
+    "propertyNames": validation_helper.propertyNames,
+    "required": validation_helper.required,
+    "type": validation_helper.type,
+    "uniqueItems": validation_helper.uniqueItems,
+}
+
+
+
+class CustomDraft7Validator(jsonschema.Draft7Validator):
+    VALIDATORS = json_schema_validator
+
+
+class DateTimeValidationModel(BaseModel):
+    datetime_validation: Optional[datetime]
+
+
+format_checker = jsonschema.FormatChecker()
+if "date-time" not in format_checker.checkers:
+
+    @format_checker.checks("date-time", ValidationError)
+    def is_date_time(instance):
+        DateTimeValidationModel(datetime_validation=instance)
+        return True
+
+def setup_lexer_validator(schema):
+    return CustomDraft7Validator(schema, format_checker=format_checker)
```

Comment:
Maybe you could allow the `CustomDraft7Validator` to be parameterised, just in case we want vanilla errors in the certain implementations